File: 01-Backend/DatajudClient.API/DatajudClient.Domain/Models/Processos/AndamentoProcesso.cs
namespace DatajudClient.Domain.Models.Processos
{
    public class AndamentoProcesso : ModelBase
    {
        public int Codigo { get; set; }
        public string Descricao { get; set; }
        public DateTime DataHora { get; set; }
        public int ProcessoId { get; set; }
        public virtual Processo Processo { get; set; }
        public virtual List<ComplementoAndamento>? Complementos { get; set; }

        public AndamentoProcesso()
        {
            Codigo = 0;
            Descricao = string.Empty;
            DataHora = DateTime.Now;
            ProcessoId = 0;
            Processo = new Processo();
            Complementos = new List<ComplementoAndamento>();
        }
    }
}

File: 02-Frontend/DatajudClient.Front/src/app/interfaces/processo/andamentoProcesso.ts
import { ComplementoAndamento } from "./complementoAndamento";

export interface AndamentoProcesso {
  Id: number;
  Codigo: number;
  Descricao: string;
  DataHora: Date;
  ComplementosDescricao: string;
  Complementos: ComplementoAndamento[] | null;
}
